types/queries.d.ts

```diff
@@ -111,24 +116,17 @@ export interface ByRoleOptions extends MatcherOptions {
   /**
    * Only considers elements with the specified accessible name.
    */
-  name?:
-    | RegExp
-    | string
-    | ((accessibleName: string, element: Element) => boolean)
+  name?: RegExp | string | MatcherFunction
   /**
    * Only considers elements with the specified accessible description.
    */
-  description?:
-    | RegExp
-    | string
-    | ((accessibleDescription: string, element: Element) => boolean)
+  description?: RegExp | string | MatcherFunction
 }
 
-export type AllByRole<T extends HTMLElement = HTMLElement> = (
-  container: HTMLElement,
-  role: ByRoleMatcher,
-  options?: ByRoleOptions,
-) => T[]
+export type AllByRole = QueryMethod<
```

Comment:
Same here. By using the generic `QueryMethod` we loose type hints. Let's type it out properly.

Reply:
Oh okay, I think QueryMethod was the right way. Do you know what are the type hints that we lose that you mention.

Reply:
Just to take in care when I work in that

Reply:
In editor suggestions:
original:
![Screenshot 2023-01-10 at 17 46 40](https://user-images.githubusercontent.com/12292047/211612041-5f6c10e0-6106-43c5-9e89-aadf6ea97a6d.png)

your change:

![Screenshot 2023-01-10 at 17 46 53](https://user-images.githubusercontent.com/12292047/211611991-c6690cd6-4f26-49e5-84a5-bd39b83da8ed.png)

Notice the name of the parameters

Reply:
ohhh!

Reply:
Thank you!